### PR TITLE
Grib

### PR DESCRIPTION
**Pull request by PeterPetrik**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ SET (MDAL_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/output CACHE PATH "Output base di
 SET (ENABLE_TESTS TRUE CACHE BOOL "Build tests?")
 SET (BUILD_TOOLS TRUE CACHE BOOL "Build tool executables?")
 SET (WITH_HDF5 TRUE CACHE BOOL "Build providers that requires HDF5")
+SET (WITH_GDAL TRUE CACHE BOOL "Build providers that requires GDAL (e.g. GRIB)")
```
Comment (wonder-sk) [inline CMakeLists.txt]: s/requires/require/ (also one line above)

File: mdal/frmts/mdal_gdal.cpp [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+  }
+}
+
+/******************************************************************************************************/
+
+bool MDAL::LoaderGdal::meshes_equals( const MDAL::GdalDataset *ds1, const MDAL::GdalDataset *ds2 ) const
+{
+  return ( ( ds1->mXSize == ds2->mXSize ) &&
+           ( ds1->mYSize == ds2->mYSize ) &&
+           ( is_nodata( ds1->mGT[0], ds2->mGT[0] ) ) &&
```
Comment (wonder-sk) [inline mdal/frmts/mdal_gdal.cpp]: can we rename it to something like fuzzy_compare? is_nodata() is very non-intuitive

File: mdal/frmts/mdal_gdal.cpp [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+        }
+      }
+      else
+      {
+        tos->values[idx].x = val;
+        tos->values[idx].noData = noData;
+      }
+    }
+
+    GDALFlushRasterCache( raster_band );
```
Comment (wonder-sk) [inline mdal/frmts/mdal_gdal.cpp]: why doing this after reading every line?

File: mdal/mdal_loader.cpp [hunk 4]
```diff
@@ -8,15 +8,31 @@
...
+  if ( !MDAL::fileExists( meshFile ) )
+  {
+    if ( status ) *status = MDAL_Status::Err_FileNotFound;
+    return nullptr;
+  }
+
   MDAL::Loader2dm loader( meshFile );
   std::unique_ptr<MDAL::Mesh> mesh = loader.load( status );
+
+#ifdef HAVE_HDF5
```
Comment (wonder-sk) [inline mdal/mdal_loader.cpp]: HAVE_GDAL?

File: mdal/mdal_utils.hpp [hunk 5]
```diff
@@ -64,5 +76,15 @@ namespace MDAL
     return ltrim( rtrim( s, delimiters ), delimiters );
   }
 
+
+  // datetime
+  class DateTime
+  {
+
+  };
```
Comment (wonder-sk) [inline mdal/mdal_utils.hpp]: Hmm?

Comment (PeterPetrik) [reply mdal/mdal_utils.hpp]: smart replacement of QDateTime :dancing_men:

**Pull request merged**